ui/org.eclipse.pde.core/src/org/eclipse/pde/internal/core/natures/PluginProject.java

```diff
@@ -36,4 +38,19 @@ public void deconfigure() throws CoreException {
 		removeFromBuildSpec(PDE.MANIFEST_BUILDER_ID);
 		removeFromBuildSpec(PDE.SCHEMA_BUILDER_ID);
 	}
+
+	public static boolean isPluginProject(IProject project) {
+		if (project.isOpen() && hasRequiredNatures(project)) {
+			return true;
+		}
+		return false;
```

Comment:
```suggestion
		return project.isOpen() && hasRequiredNatures(project);
```

Reply:
Actually the `PDE` class already has a method `hasPluginNature()` that does exactly, what this helper method does.
Therefore I suggest you just inline `project.isOpen() && PDE.hasPluginNature(project)` to the only caller or if the project actually should always be open and a not open project is an error to use `PDE.hasPluginNature(project)` only.

Reply:
It does not do exactly what this method does:

1. throw Core exception
2. do not check if project is open